src/main/java/games/strategy/net/nio/Encoder.java

```diff
@@ -90,4 +91,17 @@ private void write(final MessageHeader header, final ObjectOutputStream out, fin
     }
     out.reset();
   }
+
+  /**
+   * Provide access to the raw buffer.
+   */
+  private static final class ByteArrayOutputStream2 extends ByteArrayOutputStream {
```

Comment:
On the other hand: If you find out, that this "hack" is really not necessary, you can of course just leave the name as is, and remove this class altogether in a follow-up PR